### PR TITLE
Expanded Fastlane setup + Jenkins integration

### PR DESCRIPTION
**Pull request by sebalopez**
#### Description:

* Added test-only lane for CI use
* Minor changes to existing lanes ( ad-hoc releases for internal QA use)
* Jenkinsfiles for CI and Release pipelines
* Expanded README with complete project setup requirements (particularly regarding code signing)

### CODE REVIEW

File: ios-base.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1048,12 +1054,13 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				INFOPLIST_PREPROCESS = NO;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.2;
+				IPHONEOS_DEPLOYMENT_TARGET = 13.2;
```
Comment (glm4) [inline ios-base.xcodeproj/project.pbxproj:1057]: Please do not change this. We will update or minor deployment target but it is still in discussion and it should be in a different PR.

Comment (sebalopez) [reply ios-base.xcodeproj/project.pbxproj]: agreed, I had changed it only for testing with available simulators and should have reverted, fixed now

**Comment by germanStabile** [comment]
Amazing work @sebalopez! 

One thing we were discussing with the iOS team is that it may not make sense to incorporate the Jenkins integration to the master branch since this is open source and the implementation seems to be tightly related to Rootstrap workflow. 

We were thinking in keeping this in a feature branch were we can base our projects from. What do you think? 

That being said I really liked the additions to the fast lane documentation and I think that does belong here.   

We can discuss more in slack. Thanks!!

**Comment by sebalopez** [comment]
after internal discussion, removed the Jenkins setup from this branch, to be kept separately under  feature/jenkins

File: .gitignore [hunk 2]
```diff
@@ -65,3 +67,7 @@ fastlane/screenshots
 GoogleService*.plist
 
 R.generated.swift
+
+# Gem dependencies
+vendor/
+Gemfile.lock
```
Comment (fernandatoledo) [inline .gitignore:73]: New line at end of file is missing

File: ios-baseUITests/Info.plist [hunk 3]
```diff
@@ -17,6 +17,6 @@
 	<key>CFBundleShortVersionString</key>
 	<string>1.0</string>
 	<key>CFBundleVersion</key>
-	<string>1</string>
+	<string>1.0</string>
```
Comment (mato2593) [inline ios-baseUITests/Info.plist:20]: This should be just a number, we have `CFBundleShortVersionString` for semantic versioning.

File: README.md [hunk 4]
```diff
@@ -80,19 +81,82 @@ We strongly recommend that all private keys be added to a `.plist` file that wil
...
-Lanes for each deployment target are provided with some basic behavior:
+
+## Automated Build and Deployment using Fastlane
+
+We use [Fastlane](https://docs.fastlane.tools) to automate code signing, building and deployment. 
+
+
+### New Project Setup
+
+Several steps need to be executed before before building a project for the first time.
```
Comment (CamilaMoscatelli) [inline README.md:92]: Extra before :grin:
